### ISSUE TITLE
Feature: exif data

### ISSUE THREAD

**Issue by nayrnet** [openalpr/openalpr#344]
to improve alprd's functionality without adding dependencies for a webUI it occurs to me that json data could be dumped either the _ImageDescription_ (ASCII) or _UserComment_ (UTF) exif metadata and parsed/searched later with a simple file browser/image.

other fields could also be populated such as:
Height/Width
Name = uuid
HostComputer = hostname
Program Name = openalpr
RatingPercent = confidence
GPS = (user defined per stream)
DateTimeOriginal
ActiveArea = plate coordinates

I was thinking of implementing this at the web-poster layer, but I realized it'd be faster and more efficient I think to just save this directly to the file when it's first written.. It'd allow you to rebuild a database upon a collection of images, or extract the data with a standard image viewer remotely without a web interface and perhaps just a native app.. I think the command line application could also benifit from the feature.

comments/feedback? I'm willing to give this a crack but I dont know what existing libraries licenses may conflict with your own dual license.

**Comment by vladimirr0708982** [comment]
This can be interesting.... bat... how u will search data from image...
in any case... it's better to gave all data in some "database" 
data in image"tag" will be ok... when web GUI read image.. to read data from image...

**Comment by matthill** [comment]
Yeah, that seems like a neat idea.  I don't think this conflicts with the licensing as far as I know (unless exif format is patent protected or something like that?)  The library used for exif modification will need to be permissively licensed (i.e., Apache, MIT, BSD, etc.).

Hopefully the attributes can be added in-memory (e.g., read/write to a jpeg byte buffer) in order for it to be highly efficient.

**Comment by nayrnet** [comment]
Well on my Mac I can search through exif tags, right now (non commercial) OpenALPR has no WebUI or Database Backend, or even has the ability to really save the data.

Apps can search exif data without loading the whole file, its not as fast and efficient as a database.. especially once you get a 100k plate images, but its something at least.

**Comment by newsages** [comment]
http://www.google.com/patents/US7676118

**Comment by nayrnet** [comment]
do you think those patents are relevant? ianal, but we'd just be saving data into the standard headers and not actually transforming the image based upon the tags

and thanks for pointing me at libexif, I just did a cursory search while I was scoping this idea out and then quickly realized there may be licensing (and possibly patent) issues to navigate.

**Comment by nayrnet** [comment]
just expanding on my proposal, I like the idea of being able to archive off a bunch of images or transfer large groups of captures between people without depending on a corresponding webapp + database dump.

I've yet to start work on a way to store/search/parse this data, I dont have the time to implement what I want quickly so interim I have just been using a simple app to dump the json into a parallel _platejson_ path.. expecting at a later time to be able to parse all that into a database of my own creation.

after looking at a folder of .json files I wondered if I could just store it in the image as metadata, thus I came up with this solution to the problem. I doubt I am not the first to be in this situation.

ps: love the work so far, the results are impressive.

**Comment by matthill** [comment]
Yes, this sounds useful.  Attaching standard metadata to a jpeg doesn't take up much space and may be useful for other purposes.  I don't think that particular patent applies, so it's ok to use.
